tests/MemoryLeakWarningTest.cpp

```diff
@@ -54,6 +54,36 @@ static MemoryLeakWarningPlugin* memPlugin;
 static DummyReporter dummy;
 static TestMemoryAllocator* allocator;
 
+TEST_GROUP(MemoryLeakWarningLocalDetectorTest)
+{
+};
+
+TEST(MemoryLeakWarningLocalDetectorTest, localDetectorReturnsNewGlobalWhenNoneWasSet)
+{
+    MemoryLeakWarningPlugin memoryLeakWarningPlugin("TestMemoryLeakWarningPlugin", NULL);
+    CHECK(0 != memoryLeakWarningPlugin.getMemoryLeakDetector());
+}
+
+TEST(MemoryLeakWarningLocalDetectorTest, localDetectorIsTheOneSpecifiedInConstructor)
+{
+    MemoryLeakDetector localDetector(&dummy);
+    MemoryLeakDetector globalDetector(&dummy);
+    MemoryLeakWarningPlugin memoryLeakWarningPlugin("TestMemoryLeakWarningPlugin", &localDetector);
+    POINTERS_EQUAL(&localDetector, memoryLeakWarningPlugin.getMemoryLeakDetector());
+}
+
+IGNORE_TEST(MemoryLeakWarningLocalDetectorTest, localDetectorIsGlobalDetector)
```

Comment:
Yeah.

Not sure why setting the global detector to a local variable temporarily results in several "memory leaks" and a crash later on:
 1. A local variable is not allocated on the heap at all.
 2. Original detector and reporter are saved and restored
 3. The MemoryLeakDetectorPlugin instance goes out of scope after the test

 So, what is wrong in this picture?